test/module/libs/multihash/multihash_test.cpp

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "multihash/multihash.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include "framework/result_gtest_checkers.hpp"
+#include "multihash/type.hpp"
+#include "multihash/varint.hpp"
+
+using namespace iroha::multihash;
+using namespace shared_model::interface::types;
+
+inline ByteRange operator""_byterange(const char *c, size_t s) {
+  return ByteRange{reinterpret_cast<const std::byte *>(c), s};
+}
+
+inline std::basic_string<std::byte> operator""_bytestring(const char *c,
+                                                          size_t s) {
+  return std::basic_string<std::byte>{reinterpret_cast<const std::byte *>(c),
+                                      s};
+}
+
+static const std::initializer_list<uint64_t> kInts = {
+    0, 1, 0xF0, 0xFF, 0xFFFF, 0xFFFFFF};
+
+class VarIntTestParam : public ::testing::TestWithParam<uint64_t> {};
+
+/**
+ *   @given an integer
+ *   @when encode and decode varint
+ *   @then result is equal to former integer
+ **/
+TEST_P(VarIntTestParam, SingleIntEncDec) {
+  std::basic_string<std::byte> buffer;
+  encodeVarInt(GetParam(), buffer);
+  uint64_t read_number = 0;
+  ByteRange buffer_view{buffer.data(), buffer.size()};
+  EXPECT_TRUE(readVarInt(buffer_view, read_number));
```

Comment:
```suggestion
  EXPECT_TRUE(readVarInt(buffer, read_number));
```
:thinking:

Reply:
`readVarInt` needs lvalue reference to `ByteRange`